### PR TITLE
 Can we decouple the model with the phone code?

### PR DESCRIPTION
**Pull request by fraxedas**
The intent is to update the model separate from the app code.

### CODE REVIEW

File: SampleApps/CurrencyRecogniser/CurrencyRecogniser.Android/MainActivity.cs [hunk 1]
```diff
@@ -8,12 +8,18 @@
...
 using Plugin.CurrentActivity;
 using Xam.Plugins.OnDeviceCustomVision;
+using System.IO;
 
 namespace CurrencyRecogniser.Droid
 {
     [Activity(Label = "CurrencyRecogniser", Icon = "@mipmap/icon", Theme = "@style/MainTheme", MainLauncher = true, ConfigurationChanges = ConfigChanges.ScreenSize | ConfigChanges.Orientation)]
     public class MainActivity : global::Xamarin.Forms.Platform.Android.FormsAppCompatActivity
     {
+        bool useFilesForTheModel = false;
```
Comment (fraxedas) [inline SampleApps/CurrencyRecogniser/CurrencyRecogniser.Android/MainActivity.cs:18]: @jimbobbennett 
I added this flag to switch between assets and files for the model

File: SampleApps/CurrencyRecogniser/CurrencyRecogniser.iOS/AppDelegate.cs [hunk 2]
```diff
@@ -14,6 +15,9 @@ namespace CurrencyRecogniser.iOS
     [Register("AppDelegate")]
     public partial class AppDelegate : global::Xamarin.Forms.Platform.iOS.FormsApplicationDelegate
     {
+        bool useFilesForTheModel = false;
```
Comment (fraxedas) [inline SampleApps/CurrencyRecogniser/CurrencyRecogniser.iOS/AppDelegate.cs:18]: @jimbobbennett 
I added this flag to switch between assets and files for the model

File: Xam.Plugins.OnDeviceCustomVision/ImageClassifier.ios.cs [hunk 3]
```diff
@@ -27,7 +27,13 @@ public class ImageClassifierImplementation : ImageClassifierBase
...
         private VNCoreMLModel LoadModel(string modelName)
         {
-            var modelPath = NSBundle.MainBundle.GetUrlForResource(modelName, "mlmodelc") ?? CompileModel(modelName);
+            var isPath = File.Exists(modelName);
+            var isCompiled = Path.GetExtension(modelName).Equals("mlmodelc");
+            NSUrl modelPath = isCompiled
+                ? (isPath
+                    ? new NSUrl(modelName, false)
+                    : NSBundle.MainBundle.GetUrlForResource(modelName, "mlmodelc"))
+                : CompileModel(modelName, isPath);
```
Comment (jimbobbennett) [inline Xam.Plugins.OnDeviceCustomVision/ImageClassifier.ios.cs]: Can you clean up this code - I find nested ternary operators hard to grok, so would favor a solution with more readability.

Comment (fraxedas) [reply Xam.Plugins.OnDeviceCustomVision/ImageClassifier.ios.cs]: Done.
I changed it to nested if-else.

**Pull request merged**